src/libs/migrations/KeyReportActionsDraftByReportActionID.js

```diff
@@ -0,0 +1,59 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import Log from '../Log';
+import ONYXKEYS from '../../ONYXKEYS';
+
+/**
+ * This migration updates reportActionsDrafts data to be keyed by reportActionID.
+ *
+ * Before: reportActionsDrafts_reportID_reportActionID: value
+ * After: reportActionsDrafts_reportID: {[reportActionID]: value}
+ *
+ * @returns {Promise}
+ */
+export default function () {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS,
+            waitForCollectionCallback: true,
+            callback: (allReportActionsDrafts) => {
+                Onyx.disconnect(connectionID);
+
+                if (!allReportActionsDrafts) {
+                    Log.info('[Migrate Onyx] Skipped migration KeyReportActionsDraftByReportActionID because there were no reportActionsDrafts');
+                    return resolve();
+                }
+
+                const newReportActionsDrafts = {};
+                _.each(allReportActionsDrafts, (reportActionDraft, onyxKey) => {
+                    if (!_.isString(reportActionDraft)) {
+                        return;
+                    }
+                    newReportActionsDrafts[onyxKey] = null;
+
+                    if (_.isEmpty(reportActionDraft)) {
+                        return;
+                    }
+
+                    const reportActionID = onyxKey.split('_').pop();
+                    const newOnyxKey = onyxKey.replace(`_${reportActionID}`, '');
+                    newReportActionsDrafts[newOnyxKey] = {
+                        ...(newReportActionsDrafts[newOnyxKey] ||
+                            // Includes the existing migrated draft
+                            allReportActionsDrafts[newOnyxKey]),
+                        [reportActionID]: reportActionDraft,
+                    };
```

Comment:
This is now very hard to read, can we make it less fancy and more readable, maybe like this?
```suggestion
                    // If newReportActionsDrafts[newOnyxKey] isn't set, fall back on the migrated draft if there is one
                    const currentActionDrafts = newReportActionsDrafts[newOnyxKey] || allReportActionsDrafts[newOnyxKey];
                    newReportActionsDrafts[newOnyxKey] = {
                        ...currentActionDrafts,
                        [reportActionID]: reportActionDraft,
                    };
```